### DISCUSSION TITLE
Make a automation to start online radio stream (tune-in)

### DISCUSSION THREAD

**Discussion by casperjjordaan** [music-assistant/support#459] [Q&A] [answered]
1st all, thank you very much for this awesome new integration!

I want to start a automation at 6am to play my (already signed in and working) tune-in stream

I don't know what to select setting up the automation. Automation is not a problem for me to setup, I am just having difficulty with what to select.

Please help.

Note: my Music Assistant does already work with my Google Cast speakers (18 of them through my home) and plays tune-in when I manually start it.

Warm regards from a very exited South African HA user! 
Thank you!

**Comment by marcelveldt** [comment, marked as answer]
Within the automation editor you can select 'Play media' as 'Action type'.
Select the media player(s) you want to include in the automation and press the large 'Pick media' button below.
From there you can select Music Assistant and pick the media of your choice, such as the radio station.

<img width="1089" alt="Schermafbeelding 2022-06-19 om 23 20 41" src="https://user-images.githubusercontent.com/6389780/174500703-7f3310a6-1391-4d78-84fa-3ba29939fbf4.png">

**Comment by casperjjordaan** [comment]
Thank you very very much! Worked like a charm.

**Reply by marcelveldt** [reply]
One thing I forgot to say and is related to your other issue report/question... Make sure that you select the "mass_" prefixed entity of your players here. Although addressing the player directly will work, your other request of sending a TTS broadcast will only work on the ["mass" player](https://github.com/music-assistant/hass-music-assistant/discussions/449).

**Comment by pjdreger3** [comment]
Question about this.  When I add the action Play Media and select a player then select the Pick Media should I see Music Assistant as a Media Source ?  Right now I am seeing this :

![image](https://github.com/music-assistant/hass-music-assistant/assets/7957767/0388cd70-9b0f-4657-86fb-8a37a2b7db95)

Do I have to do something to have Music Assistant show up a  Source

**Reply by pjdreger3** [reply]
Update - I have added the add on and then tried to add the integration but I get this error when I try to add the integration 
![image](https://github.com/music-assistant/hass-music-assistant/assets/7957767/1a6b2b7b-88ae-40f6-896a-82dd47a9b959)

**Reply by OzGav** [reply]
Wrong integration version. Make sure you have show betas enabled

**Comment by kongo09** [comment]
I don't get any selection choice but only an input field asking for a Media ID.

![image](https://github.com/music-assistant/hass-music-assistant/assets/640121/26734b6b-b287-46cd-9572-86b0b3aa62cd)

Goal is to play a radio station from the radio browser. How do I find out about its Media ID?

**Reply by OzGav** [reply]
Please don’t ask the same question I two places. I have answered you on discord

**Reply by kongo09** [reply]
Thanks for the swift reply. I was trying to help by hopefully documenting the solution here where others can find it plus illustrating the visual changes that seem to indicate that the previous answers are out of date. I've seen your answer on discord but it doesn't really explain how to find the Media ID, e.g. in the case of two stations having the same name.

**Reply by OzGav** [reply]
If you have two identically named stations that is unusual but you can use the mass.search service call to find the URI

**Reply by kongo09** [reply]
The search does not produce any result:

![image](https://github.com/music-assistant/hass-music-assistant/assets/640121/a8a589f2-03a1-4fda-b192-05d186ecd032)

while inside MASS it looks like this:

![image](https://github.com/music-assistant/hass-music-assistant/assets/640121/cba99991-6f34-4ad5-91bc-61842d0194a9)

Am I doing something wrong or should I file a bug?

**Reply by OzGav** [reply]
Do you have the latest version of the integration? Try adding a limit to see if that makes a difference (could be a bug?)

Still I don’t see any duplicate names there so just use `radio ffn` as the media ID

**Reply by kongo09** [reply]
Using integration 2024.1.0 with add-on 2.0.0b92

Limit makes no difference. No results.

Calling the name works. Thanks.

**Reply by OzGav** [reply]
Still I don’t know why your search isn’t working. I get a lot of results. I assume you have the radio browser provider installed?

**Reply by kongo09** [reply]
No, TuneIn

**Reply by OzGav** [reply]
Ok well with that MA can only know about what is in your favourites so you wouldn’t get many results. Still if those stations are in the MA library I would have expected them to have appeared in your search. What if you do a search for the word Hamburg or energy do those other stations appear?

**Reply by kongo09** [reply]
Nothing appears. I would expect an empty search to return all radio stations but that also doesn't work. Search generally does work, but not for radio

**Reply by kongo09** [reply]
And playing straight works for my station `ffn` but not for my station `energy`:

![Screenshot_20240210-105621](https://github.com/music-assistant/hass-music-assistant/assets/640121/e6d011fb-d297-4367-af17-03d298765411)

**Reply by OzGav** [reply]
You can't just type energy you have to use the full station name. I was actually asking about using those words with the search. I believe I have recreated this behaviour though so I will ask the devs about it.

**Reply by kongo09** [reply]
Using the full station name didn't work either:

![Screenshot_20240210-145051](https://github.com/music-assistant/hass-music-assistant/assets/640121/0629d488-97ee-4a87-93ac-f09373280463)

Which probably brings us back to the original question: where can I find the appropriate media ID? Shouldn't it be shown in the interface somewhere?

**Reply by OzGav** [reply]
As I mentioned on Discord and is shown in the docs you just use the track, artist, album, playlist or station name. If this is ambiguous then you can further define your desired item as shown in the section `How do I get the URI?` which is in the How Do I FAQ. Additionally, as I said above you can use the mass.search service call to find the exact URI.

In relation to the above you can open an issue about that error if you can find some info in the logs. Its possibly due to the punctutation as you said `radio ffn` works. Does the Hamburg station work?
